site/documentation/content/admin-guide/hono-kafka-client-configuration.md

```diff
@@ -47,110 +54,83 @@ type configuration is ignored.
 
 The `org.eclipse.hono.client.kafka.CachingKafkaProducerFactory` factory can be used to create Kafka producers for Hono's
 Kafka based APIs. The producers created by the factory are configured with instances of the class
-`org.eclipse.hono.client.kafka.MessagingKafkaConsumerConfigProperties` which can be used to programmatically configure a producer.
+`org.eclipse.hono.client.kafka.producer.MessagingKafkaProducerConfigProperties` which can be used to programmatically
+configure a producer.
 
-The configuration needs to be provided in the form `HONO_KAFKA_PRODUCERCONFIG_${PROPERTY}` as an environment variable or
-as a Java system property in the form `hono.kafka.producerConfig.${property}`, where `${PROPERTY}` respectively
-`${property}` is any of the Kafka client's [producer properties](https://kafka.apache.org/documentation/#producerconfigs).
-The provided configuration is passed directly to the Kafka producer without Hono parsing or validating it.
+The configuration needs to be provided in the form `HONO_KAFKA_${CLIENTNAME}_PRODUCERCONFIG_${PROPERTY}` as an
+environment variable or as a Java system property in the form `hono.kafka.${clientName}.producerConfig.${property}`,
+where `${PROPERTY}` respectively `${property}` is any of the Kafka client's
+[producer properties](https://kafka.apache.org/documentation/#producerconfigs)
+and `${CLIENTNAME}` respectively `${clientName}` is the name of the client to be configured, as documented in the
+component's admin guide.
 
-The following properties can _not_ be set using this mechanism because the protocol adapters use fixed values instead 
-in order to implement the message delivery semantics defined by Hono's Telemetry and Event APIs.
+The following properties can _not_ be set because
+`org.eclipse.hono.client.kafka.producer.MessagingKafkaProducerConfigProperties` uses fixed values instead in order to
+implement the message delivery semantics defined by Hono's Telemetry and Event APIs.
 
 | Property Name     | Fixed Value |
 | :---------------- | :---------- |
 | `key.serializer`    | `org.apache.kafka.common.serialization.StringSerializer` |
 | `value.serializer`   | `io.vertx.kafka.client.serialization.BufferSerializer` |
 | `enable.idempotence` | `true` |
 
```

Comment:
How about adding a sentence here that a configured `client.id` value will only be adopted as a prefix, with the client name and a unique identifier getting attached to it? That should then be copied to the producer and admin config sections.

Reply:
Please see my changes here.